Add platform code for setting writekey (#4050)
segmentio/segment-docs: src/connections/sources/catalog/libraries/mobile/react-native/index.md

### Will I still see device-mode integrations listed as `false` in the integrations object?
When you successfully package a plugin in device-mode, you won't see the integration listed as `false` in the integrations object for a Segment event. This logic is packaged in the event metadata, and isn't surfaced in the Segment debugger.
### Why are my IDs not set in UUID format?

> Due to [limitations](https://github.com/segmentio/analytics-react-native/blob/master/packages/core/src/uuid.ts#L5){:target="_blank"} with the React Native bridge, Segment doesn't use UUID format for `anonymousId`s and `messageId`s in local development. These IDs will be set in UUID format for your live app.

Reviewer: ```suggestion
Due to [limitations](https://github.com/segmentio/analytics-react-native/blob/master/packages/core/src/uuid.ts#L5){:target="_blank"} with the React Native bridge, Segment doesn't use UUID format for `anonymousId` and `messageId` values in local development. These IDs will be set in UUID format for your live app.
```

### How do I set a distinct writeKey for iOS and android?

> You may wish to set a distinct writeKey for iOS and android, this could be used to send data to different destinations depending on the client side platform. To achieve this, you can dynamically set the writeKey when initializing the Segment client:

Reviewer: ```suggestion
You can set different writeKeys for iOS and Android. This is helpful if you want to send data to different destinations based on the client side platform. To set different writeKeys, you can dynamically set the writeKey when you initialize the Segment client:
```
